docs/docsite/rst/user_guide/intro_inventory.rst

```diff
@@ -478,6 +478,96 @@ Here is an example of how to instantly deploy to created containers::
+To avoid copy pasting the commmon groups and variables the following directory layout is created::
+
+    inventories/
+       production/
+          hosts               # inventory file for production servers
+          group_vars/
+             db-servers.yml   # assign variables to db servers in production environment
+       staging
+          hosts               # inventory file for staging environment
+       common/
+          hosts               # inventory file for defining common groups for all environments
+          group_vars/
+             all-servers.yml  # assign variables to all servers in all environments
+
+The production inventory is defined like this:
+
+.. code-block:: guess
+
+    [web-servers]
+    web1.example.com
+
+    [db-servers]
+    db1.example.com
+
+The staging environment is very similar:
+
+.. code-block:: guess
+
+    [web-servers]
+    web1.staging.example.com
+
+    [db-servers]
+    db1.staging.example.com
+
+The common inventory composes a new group named all-servers:
+
+.. code-block:: guess
+
+    [web-servers]
+
+    [db-servers]
+
+    [all-servers:children]
+    web-servers
+    db-servers
+
+The inventories are merged, so when you target common and production
+inventories all-servers group will have web1.example.com and db1.example.com as members and
+db1.example.com will have host variables from both db-servers.yml and all-servers.yml.
+In case of conflict a variable defined in db-servers.yml wins over all-servers.yml's variable.
+
+Without the empty web-servers and db-servers groups the common inventory would have a dependency on those
+groups and would need to be read after the production inventory. This would make it harder for the 
+production inventory to override the common group variables.
+
+You can target these inventories like this (note the order of the inventory parameters)::
+
+    ansible-playbook example.yml -i inventories/common -i inventories/production
+
+It is also possible to configure :envvar:`ANSIBLE_INVENTORY` to use multiple inventories.
+
+**Aggregating inventory sources with a directory**
+
+You can combine multiple inventory sources under a directory. As an example the following inventory
+combines an inventory plugin source, a dynamic inventory script, and files with static hosts and groups::
+
+    inventory/
+      01-openstack.yml          # configure inventory plugin to get hosts from Openstack cloud
+      02-dynamic-inventory.py   # add additional hosts with dynamic inventory script
+      03-static-inventory       # add static hosts
+      04-static-groups          # define static groups using groups from earlier inventory sources as children
+      group_vars/
+        all.yml                 # assign variables to all hosts
+
+The inventories are parsed in alphabetical order so number prefixes are used to make sure
+the parsing order is correct. For more details about inventory plugins see :ref:`inventory_plugins`.
```

Comment:
Why would I want to control the parsing order in this particular way? What happens if I get this wrong?

Reply:
parsing order can control conflicts with variables, if you define a variable in a group twice with different values, last update wins. 

i.e  in the `[all:vars]`  in01_stuff.ini defines `myvar=1`, but 03_stuff.ini defines `myvar=2`, so in the end `myvar == 2`